osu.Game/Screens/Select/BeatmapCarousel.cs

```diff
@@ -232,6 +232,8 @@ public void SelectPreviousRandom()
 
         public bool AllowSelection = true;
 
+        public bool PendingFilter => filterTask?.Completed == false;

```

Comment:
Oh, I didn't realize this logic was spread over 2 classes. I'd prefer a `FlushPendingFilters` method so that none of all this needs to be exposed.